devicesService.go

```
@@ -29,3 +31,14 @@ func (ds *DeviceService) Get(id int) (*DeviceWithId, error) {
 	}
 	return d, nil
 }
+
+func (ds *DeviceService) Get(limit int, page int) (*[]DeviceWithId, error) {
+	if limit < 0 || page < 0 {
+		return nil, NewBadRequestError("negative limit or page")
+	}
+	d, err := ds.DeviceRepo.Get(limit, page)
+	if err != nil {
+		return nil, NewInternalServerError("database error")
```

Since you are swallowing here original error, you should log it here. It will matter when you will start with mongo or influx